sirepo/template/hdf5_util.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""HDF5 utilities.
+
+:copyright: Copyright (c) 2023 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern.pkcollections import PKDict
+from pykern.pkdebug import pkdp
+from sirepo.template import template_common
+import numpy as np
+
+
+class HDF5Util:
+    # TODO(rorour): add docs for heatmap & lineplot
+    def __init__(self):
+        pass
+
+    def heatmap(self, plot_attrs):
+        x_field = "x"
+        y_field = "y"
+        plots = PKDict()
+        for f in (x_field, y_field):
+            n = plot_attrs.model[f]
+            plots[f] = PKDict(
+                name=n,
+                points=np.array(
+                    plot_attrs.file["/Step#{}/{}".format(plot_attrs.frame_index, n)]
+                ),
+                label=n,
+            )
+            plot_attrs.format_plot(plot_attrs.file, plots[f])
+
+        return template_common.heatmap(
+            values=[plots[x_field].points, plots[y_field].points],
+            model=plot_attrs.model,
+            plot_fields=PKDict(
+                x_label=plots[x_field].label,
+                y_label=plots[y_field].label,
+                title=plot_attrs.title,
+            ),
+        )
+
+    def lineplot(self, plot_attrs):
+        x_field = "x"
+        y_fields = ("y1", "y2", "y3")
+
+        plots = PKDict()
+        for f in (x_field,) + y_fields:
+            name_and_field = plot_attrs.model[f].split(" ")
+            if name_and_field[0] == "none":
+                continue
+            plots[f] = PKDict(
+                label=plot_attrs.model[f],
+                dim=f,
+                points=[],
+                name=name_and_field[0],
+                index=plot_attrs.index(name_and_field),
+            )
+        plot_attrs.format_plots(plots)
+        x = plots.get(x_field)
```

Comment:
here format_plots() could be passed the open file handle as an argument, so the caller doesn't need to open the hdf5 file beforehand.

Reply:
I'm not totally understanding your vision here- let me know if this still needs to change